src/ML/include/BipedalLocomotion/ML/velMANNAutoregressive.h

```diff
@@ -0,0 +1,334 @@
+    Eigen::Matrix2Xd desiredFutureBaseDirections;
+
+    /** Matrix containing the desired base velocity trajectory. The rows contain the x and y
+     * velocity projected into the ground while the columns the position at each time instant. */
+    Eigen::Matrix2Xd desiredFutureBaseVelocities;
+
+    /** Matrix containing the desired base angular velocity trajectory around the z axis (yaw). The columns denote the position at each time instant. */
+    Eigen::RowVectorXd desiredFutureBaseAngVelocities;
+};
+
+/**
+ * velMANNAutoregressiveOutput contains the output to the velMANN network when used in autoregressive
+ * fashion.
+ */
+struct velMANNAutoregressiveOutput
+{
+    Eigen::VectorXd jointsPosition; /**< Joint positions in radians */
+    Eigen::VectorXd jointsVelocity; /**< Joint velocities in radians */
+    manif::SE3d basePose; /**< Base pose with respect to the inertial frame, i.e., \f${}^I H_B\f$ */
+    manif::SE3d::Tangent baseVelocity; /**< Base velocity in mixed representation */
+    Eigen::Vector3d comPosition; /**< Position of the CoM with respect to the inertial frame */
+    Eigen::Vector3d comVelocity; /**< Time derivative of the CoM position with respect to the inertial frame */
+    Eigen::Vector3d angularMomentum; /**< Centroidal angular momentum */
+    manif::SE3d leftFootPose; /**< Left foot pose with respect to the inertial frame, i.e., \f${}^I H_B\f$ */
+    manif::SE3d rightFootPose; /**< Right foot pose with respect to the inertial frame, i.e., \f${}^I H_B\f$ */
+    Eigen::VectorXd leftFootVelocity; /**< Left foot velocity with respect to the inertial frame, i.e., \f${}^I H_B\f$ */
+    Eigen::VectorXd rightFootVelocity; /**< Right foot velocity with respect to the inertial frame, i.e., \f${}^I H_B\f$ */
+    Contacts::EstimatedContact leftFoot; /**< Left foot contact */
+    Contacts::EstimatedContact rightFoot; /**< Right foot contact */
+    std::chrono::nanoseconds currentTime; /**< Current time stored in the advanceable */
+};
+
+// clang-format off
+/**
+ * velMANNAutoregressive is a class that allows to perform autoregressive inference with Mode-Adaptive
+ * Neural Networks (MANN).
+ * @subsection velMANN_autoregressive velMANN Autoregressive
+ * The following diagram shows how the velMANN network is used inside the velMANNAutoregressive class.
+ * The output of velMANN is given to two blocks, one computes the kinematically feasible base position,
+ * while the other blends it to te input provided by the user
+ * <img src="https://user-images.githubusercontent.com/16744101/237064989-b3d0d62f-2fc9-4ca4-9b11-ecca0c8c378e.png" alt="mann_autoregressive">
+ * @note The implementation of the class follows the work presented in "P. M. Viceconte et al.,
+ * "ADHERENT: Learning Human-like Trajectory Generators for Whole-body Control of Humanoid Robots,"
+ * in IEEE Robotics and Automation Letters, vol. 7, no. 2, pp. 2779-2786, April 2022,
+ * doi: 10.1109/LRA.2022.3141658." https://doi.org/10.1109/LRA.2022.3141658
```

Comment:
You may avoid adding the picture here since it refers to a different implementation of MANN. Moreover you can link your paper

Reply:
Done in 8bf5fe24.